sys/include/net/gnrc/netif2/ipv6.h

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup net_gnrc_netif2
+ * @{
+ *
+ * @file
+ * @brief   IPv6 defintions for @ref net_gnrc_netif2
+ *
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+#ifndef NET_GNRC_NETIF2_IPV6_H
+#define NET_GNRC_NETIF2_IPV6_H
+
+#include "evtimer_msg.h"
+#include "net/ipv6/addr.h"
+#ifdef MODULE_GNRC_IPV6_NIB
+#include "net/gnrc/ipv6/nib/conf.h"
+#endif
+#include "net/gnrc/netif2/conf.h"
+#ifdef MODULE_NETSTATS_IPV6
+#include "net/netstats.h"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    IPv6 unicast and anycast address flags
+ * @anchor  net_gnrc_netif2_ipv6_addrs_flags
+ * @{
+ */
+/**
+ * @brief   Mask for the address' state
+ */
+#define GNRC_NETIF2_IPV6_ADDRS_FLAGS_STATE_MASK             (0x1fU)
+
+/**
+ * @brief   Tentative states (with encoded DAD retransmissions)
+ *
+ * The retransmissions of DAD transmits can be decoded from this state by
+ * applying it as a mask to the [flags](gnrc_netif2_ipv6_t::addrs_flags) of the
+ * address.
+ */
+#define GNRC_NETIF2_IPV6_ADDRS_FLAGS_STATE_TENTATIVE        (0x07U)
+
+/**
+ * @brief   Deprecated address state (still valid, but not preferred)
+ */
+#define GNRC_NETIF2_IPV6_ADDRS_FLAGS_STATE_DEPRECATED       (0x08U)
+
+/**
+ * @brief   Valid address state
+ */
+#define GNRC_NETIF2_IPV6_ADDRS_FLAGS_STATE_VALID            (0x10U)
+
+/**
+ * @brief   Address is an anycast address
+ */
+#define GNRC_NETIF2_IPV6_ADDRS_FLAGS_ANYCAST                (0x20U)
+/** @} */
+
+/**
+ * @brief   IPv6 component for @ref gnrc_netif2_t
+ *
+ * @note only available with @ref net_gnrc_ipv6.
+ */
+typedef struct {
+    /**
+     * @brief   Flags for gnrc_netif2_t::ipv6_addrs
+     *
+     * @see net_gnrc_netif2_ipv6_addrs_flags
+     *
+     * @note    Only available with module @ref net_gnrc_ipv6 "gnrc_ipv6".
```

Comment:
>Only available with module @ref net_gnrc_ipv6 "gnrc_ipv6".

shouldn't there be a guard around those struct members if they are only available with `gnrc_ipv6`?

Reply:
There is, not just around the single struct members, but [around the whole struct](https://github.com/miri64/RIOT/blob/173756a5d1536253e4b4709f0b2ce738afda3526/sys/include/net/gnrc/netif2.h#L65-L67) ;-).